go/test/endtoend/cluster/vttablet_process.go

```diff
@@ -0,0 +1,207 @@
+		"-backup_storage_implementation", vttablet.BackupStorageImplementation,
+		"-file_backup_storage_root", vttablet.FileBackupStorageRoot,
+		"-restore_from_backup",
+		"-service_map", vttablet.ServiceMap,
+		"-vtctld_addr", vttablet.VtctldAddress,
+	)
+	vttablet.proc.Args = append(vttablet.proc.Args, vttablet.ExtraArgs...)
+
+	vttablet.proc.Stderr = os.Stderr
+	vttablet.proc.Stdout = os.Stdout
+
+	vttablet.proc.Env = append(vttablet.proc.Env, os.Environ()...)
+
+	log.Infof("%v %v", strings.Join(vttablet.proc.Args, " "))
+
+	err = vttablet.proc.Start()
+	if err != nil {
+		return
+	}
+
+	vttablet.exit = make(chan error)
+	go func() {
+		vttablet.exit <- vttablet.proc.Wait()
+	}()
+
+	timeout := time.Now().Add(60 * time.Second)
+	for time.Now().Before(timeout) {
+		if vttablet.WaitForStatus("NOT_SERVING") {
+			return nil
+		}
+		select {
+		case err := <-vttablet.exit:
+			return fmt.Errorf("process '%s' exited prematurely (err: %s)", vttablet.Name, err)
+		default:
+			time.Sleep(300 * time.Millisecond)
+		}
+	}
+
+	return fmt.Errorf("process '%s' timed out after 60s (err: %s)", vttablet.Name, <-vttablet.exit)
+}
+
+// WaitForStatus function checks if vttablet process is up and running
+func (vttablet *VttabletProcess) WaitForStatus(status string) bool {
+	resp, err := http.Get(vttablet.VerifyURL)
+	if err != nil {
+		return false
+	}
+	if resp.StatusCode == 200 {
+		resultMap := make(map[string]interface{})
+		respByte, _ := ioutil.ReadAll(resp.Body)
+		err := json.Unmarshal(respByte, &resultMap)
+		if err != nil {
+			panic(err)
+		}
+		return resultMap["TabletStateName"] == status
+	}
+	return false
+}
+
+// TearDown shuts down the running vttablet service
+func (vttablet *VttabletProcess) TearDown(keepdata bool) error {
+	if vttablet.proc == nil {
+		fmt.Printf("No process found for vttablet %d", vttablet.TabletUID)
+	}
+	if vttablet.proc == nil || vttablet.exit == nil {
+		return nil
+	}
+	// Attempt graceful shutdown with SIGTERM first
+	vttablet.proc.Process.Signal(syscall.SIGTERM)
+
+	if !*keepData {
+		_ = os.RemoveAll(vttablet.Directory)
+	}
+
+	select {
+	case err := <-vttablet.exit:
+		vttablet.proc = nil
+		return err
+
+	case <-time.After(10 * time.Second):
+		vttablet.proc.Process.Kill()
+		vttablet.proc = nil
+		return <-vttablet.exit
+	}
+}
+
+// VttabletProcessInstance returns a VttabletProcess handle for vttablet process
+// configured with the given Config.
+// The process must be manually started by calling setup()
+func VttabletProcessInstance(port int, grpcPort int, tabletUID int, cell string, shard string, keyspace string, vtctldPort int, tabletType string, topoPort int, hostname string, tmpDirectory string, extraArgs []string) *VttabletProcess {
+	vtctl := VtctlProcessInstance(topoPort, hostname)
+	vttablet := &VttabletProcess{
+		Name:                        "vttablet",
+		Binary:                      "vttablet",
+		FileToLogQueries:            path.Join(tmpDirectory, fmt.Sprintf("/vt_%010d/vttable.pid", tabletUID)),
```

Comment:
This should not be the pid file, it should be some other file (e.g. querylog.txt)

Reply:
done